compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -387,6 +387,24 @@ pub(crate) unsafe fn create_module<'ll>(
         }
     }
 
+    match (sess.opts.unstable_opts.small_data_threshold, sess.target.small_data_threshold_support())
+    {
+        // Set up the small-data optimization limit for architectures that use
+        // an LLVM module flag to control this.
+        (Some(threshold), SmallDataThresholdSupport::LlvmModuleFlag(flag)) => {
+            let flag = format!("{flag}\0");
```

Comment:
There is a `CString` for that, no need to manually add nulls.

Reply:
Something like `let flag = CString::new(flag).unwrap()`? Or should `SmallDataThresholdSupport::LllvmModuleFlag` contain a `StaticCow<Cstr>` rather than `StaticCow<str>`?

Reply:
No, changing type to `StaticCow<Cstr>` is probably no needed.

Reply:
OK, tweaked to use `let flag = SmallCStr::new(flag.as_ref());`